src/main/java/duke/commands/AddCommand.java

```diff
@@ -0,0 +1,66 @@
+package duke.commands;
+
+import duke.*;
+import duke.tasks.*;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+
+public class AddCommand extends Command {
+    private static final String AT_DELIMITER = "/at";
+    private static final String BY_DELIMITER = "/by";
+    private static final String TODO_COMMAND = "todo";
+    private static final String EVENT_COMMAND = "event";
+    private static final String DEADLINE_COMMAND = "deadline";
+    private static final String ADDED_TASK_STRING = "Got it. I've added this task:";
+    private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("d/M/yyyy HHmm");
+
+    private String input;
+
+    public AddCommand(String input) {
+        super();
+        this.input = input;
+    }
+
+    public void execute(Storage storage, Ui ui, TaskList tasklist) {
+        String[] args = input.split(" ");
+        Task newTask;
+        try {
+            switch (args[0]) {
```

Comment:
You can consider creating subclasses that inherit from the AddCommand (specifically, one for each of the deadline, event and todo task types)